### ISSUE TITLE
Pre-trained model using triplet loss?

### ISSUE THREAD

**Issue by ba305** [KaiyangZhou/deep-person-reid#185]
Hi, I am curious about whether there are any pre-trained models that use triplet loss in the Model Zoo. As far as I can tell, all the Model Zoo examples use softmax loss.

Also, if I want to use triplet loss, do you have any suggestions about important hyperparameters to change in the "python train.py" call? Such as max_epoch, etc., which might be different for softmax vs. triplet loss. I thought I would ask in case you had some previous experience with this.

 Thank you!

**Comment by ba305** [comment]
Never mind, I think I have figured out the most important flags to train a triplet loss model. 

For future reference, if anyone else is curious, some important things to change are:

--loss triplet
--train-sampler RandomIdentitySampler

Also, these should already be the defaults so you probably don't need to do anything, but just make sure to have --weight-t 1, and --weight-x 0 (to only use triplet loss, not softmax loss).

And it also probably helps to increase the batch size (I increased it until my GPU memory was full) so that the triplet mining process can find harder (i.e. more difficult) triplets (although I have NOT studied this extensively). Also, it is of course a good idea to experiment with the number of epochs, learning rate, learning rate scheduler, etc.

Thanks again for providing this great repo. Please feel free to close this issue unless you have anything else you want to add.
